yt_dlp/extractor/pandatv.py

```diff
@@ -0,0 +1,68 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    UserNotLive,
+    int_or_none,
+    traverse_obj,
+    unified_strdate,
+    url_or_none,
+    urlencode_postdata,
+)
+
+
+class PandatvLiveIE(InfoExtractor):
+    _VALID_URL = r'(?P<base_url>https?://(?:www\.|m\.)?pandalive\.co\.kr)/play/(?P<id>[\da-z]+)'
+    _TESTS = [{
+        'url': 'https://www.pandalive.co.kr/play/bebenim',
+        'info_dict': {
+            'id': 'bebenim',
+            'ext': 'mp4',
+            'channel': '릴리ෆ',
+            'title': r're:앙앙❤ \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
+            'thumbnail': r're:https://cdn\.pandalive\.co\.kr/ivs/v1/.+/thumb.jpg',
+            'concurrent_view_count': int,
+            'like_count': int,
+            'live_status': 'is_live',
+            'upload_date': str,
+        },
+        'skip': 'The channel is not currently live',
+    }]
+
+    def _real_extract(self, url):
+        base_url, channel_id = self._match_valid_url(url).groups()
+        http_headers = {'Origin': base_url}
+
+        # Prepare POST data
+        post_data = {'action': 'watch', 'userId': channel_id, 'password': '', 'shareLinkType': ''}
+        post_data_bytes = urlencode_postdata(post_data)
+
+        # Fetch video metadata
+        video_meta = self._download_json(
+            'https://api.pandalive.co.kr/v1/live/play', channel_id, 'Downloading video meta data',
+            errnote=' Unable to download video meta data', data=post_data_bytes, expected_status=(200, 400))
```

Comment:
- we can (and should) inline the POST data
- you don't need to include `200` in the `expected_status` tuple; it is implicit
- are there password-protected streams?
- these comments are unnecessary
```suggestion
        video_meta = self._download_json(
            'https://api.pandalive.co.kr/v1/live/play', channel_id,
            'Downloading video meta data', 'Unable to download video meta data',
            data=urlencode_postdata({
                'action': 'watch',
                'userId': channel_id,
                'password': '',
                'shareLinkType': '',
            }), expected_status=400)
```